e2e/src/tests/promptExportImport.test.ts

```diff
@@ -430,12 +453,303 @@ test('Import file from 1.4 version to prompts and continue working with it', asy
+      await localStorageManager.setPrompts(...nestedPrompts);
+      await localStorageManager.setOpenedFolders(...nestedFolders);
+    });
+
+    await test.step('Export prompt from 3rd level folder', async () => {
+      await dialHomePage.openHomePage();
+      await dialHomePage.waitForPageLoaded({ isNewConversationVisible: true });
+      await folderPrompts.openFolderPromptDropdownMenu(
+        nestedFolders[levelsCount].name,
+        nestedPrompts[levelsCount].name,
+      );
+      exportedData = await dialHomePage.downloadData(() =>
+        promptDropdownMenu.selectMenuOption(MenuOptions.export),
+      );
+    });
+
+    await test.step('Delete all prompts and folders, import exported prompt and verify folder structure with 3rd level prompt are displayed', async () => {
+      await promptBar.deleteAllPrompts();
+      await confirmationDialog.confirm();
+      await dialHomePage.uploadData(exportedData, () =>
+        promptBar.importButton.click(),
+      );
+
+      for (const nestedFolder of nestedFolders) {
+        await folderPrompts.getFolderByName(nestedFolder.name).waitFor();
+      }
+
+      await folderPrompts
+        .getFolderPrompt(
+          nestedFolders[levelsCount].name,
+          nestedPrompts[levelsCount].name,
+        )
+        .waitFor();
+
+      expect
+        .soft(
+          await folderPrompts.getFolderPromptsCount(),
+          ExpectedMessages.promptsCountIsValid,
+        )
+        .toBe(1);
+    });
+
+    await test.step('Delete last folder with its prompt, re-import exported file again and verify last nested folder with its prompt imported', async () => {
+      await folderPrompts.openFolderDropdownMenu(
+        nestedFolders[levelsCount].name,
+      );
+      await folderDropdownMenu.selectMenuOption(MenuOptions.delete);
+      await confirmationDialog.confirm();
+
+      await dialHomePage.uploadData(exportedData, () =>
+        promptBar.importButton.click(),
+      );
+
+      await folderPrompts
+        .getFolderPrompt(
+          nestedFolders[levelsCount].name,
+          nestedPrompts[levelsCount].name,
+        )
+        .waitFor();
+    });
+
+    await test.step('Delete 2nd level folder with its nested content, re-import exported file and verify 2nd/3rd level folders with 3rd level prompt are imported', async () => {
+      await folderPrompts.openFolderDropdownMenu(
+        nestedFolders[levelsCount - 1].name,
+      );
+      await folderDropdownMenu.selectMenuOption(MenuOptions.delete);
+      await confirmationDialog.confirm();
+
+      await dialHomePage.uploadData(exportedData, () =>
+        promptBar.importButton.click(),
+      );
+
+      await folderPrompts
+        .getFolderPrompt(
+          nestedFolders[levelsCount].name,
+          nestedPrompts[levelsCount].name,
+        )
+        .waitFor();
+
+      await folderPrompts
+        .getFolderByName(nestedFolders[levelsCount - 1].name)
+        .waitFor();
```

Comment:
@irinakartun , why do you use `waitFor()` here and in other places if you already have `await`? does it convert to async-method?

Reply:
it simply waits for element visibility on the page